File: src/utils/initialState/characters.ts
import type { Actions } from '@nanosh/types/actions'
import type {
  Character,
  CharacterEquipmentSlots,
  CharacterNames,
} from '@nanosh/types/character'
import { ITEM_ID_LENGTH, MAX_AP } from '@nanosh/types/generic'
import type {
  ModifierTracker,
  ModifiersCharacter,
} from '@nanosh/types/modifiers'
import type { ShipCargo } from '@nanosh/types/ship'
import type { Skills } from '@nanosh/types/skills'
import type { Traits } from '@nanosh/types/traits'
import seedrandom from 'seedrandom'
import GetRandomString from '../getRandomString'

export const GetInitialCharacters = (
  gameID: string,
  invokeTime: number,
): Map<CharacterNames, Character> => {
  const prng = seedrandom(`${gameID}-${invokeTime}`)

  return new Map<CharacterNames, Character>([
    [
      'Solas Mercer',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.legatus-legionis']),
        skills: new Set<Skills>([
          'skill.triarii',
          'skill.tactician',
          'skill.experienced-commander',
          'skill.visionary',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.unique.guns.p2075',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Momo Tzigane',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.defender']),
        skills: new Set<Skills>([
          'skill.infantry',
          'skill.protector',
          'skill.triarii',
          'skill.stalwart',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.guns.principes',
              broken: false,
            },
          ],
          [
            'body',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'body.heavy.lorica',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Val',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.hardworking']),
        skills: new Set<Skills>([
          'skill.technician',
          'skill.engineer',
          'skill.artisan',
          'skill.diligent',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.unique.vigiles-45',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.omni-converter',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Brianne "Bree" Cheeseworth',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.gung-ho']),
        skills: new Set<Skills>([
          'skill.pirate',
          'skill.agile',
          'skill.mechpilot',
          'skill.engineer',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.guns.rondel',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Niral Pierce',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.meticulous']),
        skills: new Set<Skills>([
          'skill.surgeon',
          'skill.physician',
          'skill.pharmacologist',
          'skill.savant',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'meds.vigorisk',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      `Tee'elise "Teal" Qing`,
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.amorous']),
        skills: new Set<Skills>([
          'skill.technician',
          'skill.engineer',
          'skill.artisan',
          'skill.diligent',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>([
          {
            id: GetRandomString(ITEM_ID_LENGTH, prng),
            itemName: 'meds.somnoxa',
            broken: false,
          },
        ]),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.guns.rondel',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'X7-Gastronia "Gass" Petalnova',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.droid']),
        skills: new Set<Skills>([
          'skill.botanist',
          'skill.cook',
          'skill.engineer',
          'skill.demolition',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Ysara Mercer',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.driven']),
        skills: new Set<Skills>([
          'skill.prodigy-leader',
          'skill.aviator',
          'skill.shooter',
          'skill.tactician',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.guns.pugio',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Zedius Windsor',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.calm']),
        skills: new Set<Skills>([
          'skill.technician',
          'skill.protector',
          'skill.mechpilot',
          'skill.stalwart',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Viero Alden',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.ace']),
        skills: new Set<Skills>([
          'skill.aviator',
          'skill.guns-fundamental',
          'skill.stalwart',
          'skill.visionary',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.guns.pugio',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Soren Koda',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.dutiful']),
        skills: new Set<Skills>([
          'skill.sniper',
          'skill.guns-fundamental',
          'skill.diligent',
          'skill.adaptable',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>([]),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'weapon',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'weapon.guns.pugio',
              broken: false,
            },
          ],
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Alisa Huang',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.popular']),
        skills: new Set<Skills>([
          'skill.comms-savvy',
          'skill.persuasion',
          'skill.logistician',
          'skill.cook',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
    [
      'Rina Mikami',
      {
        ap: 0,
        maxAP: MAX_AP,
        trait: new Set<Traits>(['trait.regenesis']),
        skills: new Set<Skills>([
          'skill.silver',
          'skill.adaptable',
          'skill.savant',
          'skill.visionary',
        ]),
        modifiers: new Map<ModifiersCharacter, ModifierTracker>(),
        playerID: '',
        cycleActions: new Map<number, Actions>(),
        inventory: new Set<ShipCargo>(),
        equipment: new Map<CharacterEquipmentSlots, ShipCargo>([
          [
            'acc-1',
            {
              id: GetRandomString(ITEM_ID_LENGTH, prng),
              itemName: 'acc.voxlink',
              broken: false,
            },
          ],
        ]),
        location: 'dorms',
      },
    ],
  ])
}

File: src/nakama/gamelogic/utils/initial/characters.go
package initial

import (
	"nanosh/nakama-modules/gamelogic/types"
	"nanosh/nakama-modules/gamelogic/utils/random"
)

// GetInitialCharacters creates and returns all 13 initial characters for the game
func GetInitialCharacters(gameID string, invokeTime int64, username string) map[types.CharacterName]*types.Character {
	// Create deterministic PRNG for item IDs
	prng := random.CreateSeededPRNG(gameID, invokeTime, username)

	characters := make(map[types.CharacterName]*types.Character)

	// Solas Mercer - Leader
	characters[types.CharacterSolasMercer] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitLegatusLegionis: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillTriarii:              true,
			types.SkillTactician:            true,
			types.SkillExperiencedCommander: true,
			types.SkillGunsFundamental:      true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponUniqueGunsP2075,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Momo Tzigane - Tank
	characters[types.CharacterMomoTzigane] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitDefender: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillInfantry:  true,
			types.SkillProtector: true,
			types.SkillTriarii:   true,
			types.SkillStalwart:  true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponGunsPrincipes,
				Broken:   false,
			},
			types.EquipmentSlotBody: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemBodyHeavyLorica,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Val - Engineer
	characters[types.CharacterVal] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitHardworking: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillTechnician: true,
			types.SkillEngineer:   true,
			types.SkillArtisan:    true,
			types.SkillDiligent:   true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponUniqueVigiles45,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccOmniConverter,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Brianne "Bree" Cheeseworth - Pilot
	characters[types.CharacterBrianneCheeseworth] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitGungHo: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillPirate:          true,
			types.SkillAgile:           true,
			types.SkillMechpilot:       true,
			types.SkillGunsFundamental: true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponGunsRondel,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Niral Pierce - Doctor
	characters[types.CharacterNiralPierce] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitMeticulous: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillSurgeon:        true,
			types.SkillPhysician:      true,
			types.SkillPharmacologist: true,
			types.SkillSavant:         true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory: map[string]*types.ShipCargo{
			random.GetRandomString(types.ITEM_ID_LENGTH, prng): {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemMedsVigorisk,
				Broken:   false,
			},
		},
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Tee'elise "Teal" Qing - Combat Medic
	characters[types.CharacterTealQing] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitAmorous: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillCombatMedic: true,
			types.SkillPhysician:   true,
			types.SkillAgile:       true,
			types.SkillInfantry:    true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory: map[string]*types.ShipCargo{
			random.GetRandomString(types.ITEM_ID_LENGTH, prng): {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemMedsSomnoxa,
				Broken:   false,
			},
		},
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponGunsRondel,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// X7-Gastronia "Gass" Petalnova - Droid Chef
	characters[types.CharacterGassPetalnova] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitDroid: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillBotanist:   true,
			types.SkillCook:       true,
			types.SkillEngineer:   true,
			types.SkillDemolition: true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Ysara Mercer - Ace Pilot
	characters[types.CharacterYsaraMercer] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitDriven: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillProdigyLeader:   true,
			types.SkillAviator:         true,
			types.SkillGunsFundamental: true,
			types.SkillTactician:       true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponGunsPugio,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Zedius Windsor - Support
	characters[types.CharacterZediusWindsor] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitCalm: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillTechnician: true,
			types.SkillProtector:  true,
			types.SkillMechpilot:  true,
			types.SkillStalwart:   true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Viero Alden - Ace
	characters[types.CharacterVieroAlden] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitAce: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillAviator:         true,
			types.SkillGunsFundamental: true,
			types.SkillStalwart:        true,
			types.SkillVisionary:       true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponGunsPugio,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Soren Koda - Sniper
	characters[types.CharacterSorenKoda] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitDutiful: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillSniper:          true,
			types.SkillGunsFundamental: true,
			types.SkillDiligent:        true,
			types.SkillAdaptable:       true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotWeapon: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemWeaponGunsPugio,
				Broken:   false,
			},
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Alisa Huang - Diplomat
	characters[types.CharacterAlisaHuang] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitPopular: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillCommsSavvy:  true,
			types.SkillPersuasion:  true,
			types.SkillLogistician: true,
			types.SkillCook:        true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	// Rina Mikami - Silver
	characters[types.CharacterRinaMikami] = &types.Character{
		AP:    0,
		MaxAP: types.MAX_AP,
		Trait: map[types.Trait]bool{
			types.TraitRegenesis: true,
		},
		Skills: map[types.Skill]bool{
			types.SkillSilver:    true,
			types.SkillAdaptable: true,
			types.SkillSavant:    true,
			types.SkillVisionary: true,
		},
		Modifiers:    make(map[types.ModifierCharacter]*types.ModifierTracker),
		PlayerID:     "",
		CycleActions: make(map[int]types.Action),
		Inventory:    make(map[string]*types.ShipCargo),
		Equipment: map[types.CharacterEquipmentSlot]*types.ShipCargo{
			types.EquipmentSlotAcc1: {
				ID:       random.GetRandomString(types.ITEM_ID_LENGTH, prng),
				ItemName: types.ItemAccVoxlink,
				Broken:   false,
			},
		},
		Location: types.RoomDorms,
	}

	return characters
}
